arduino/libraries/librariesresolver/cpp.go

```diff
@@ -192,6 +192,8 @@ func ComputePriority(lib *libraries.Library, header, arch string) int {
 		priority += 600
 	} else if name == header || dirName == header {
 		priority += 500
+	} else if name == header+"-main" || dirName == header+"-main" {
+		priority += 410
```

Comment:
We are using the "tens" for bumping core-optimized libraries, so I would just upgrade the others like this:

```suggestion
		priority += 700
	} else if name == header || dirName == header {
		priority += 600
	} else if name == header+"-main" || dirName == header+"-main" {
		priority += 500
```